flow/CompressionUtils.cpp

```diff
@@ -94,59 +104,82 @@ StringRef CompressionUtils::decompress(const CompressionFilter filter, const Str
+	const int size = deterministicRandom()->randomInt(512, 1024);
+	std::string s(size, 'x');
+	Standalone<StringRef> uncompressed = Standalone<StringRef>(StringRef(s));
+	printf("Size before: %d\n", (int)uncompressed.size());
 
-	return Void();
+	Standalone<StringRef> compressed = CompressionUtils::compress(filter, uncompressed, arena);
+	ASSERT_NE(compressed.compare(uncompressed), 0);
+	printf("Size after: %d\n", (int)compressed.size());
+	// Assert compressed size is less than half.
+	ASSERT(compressed.size() * 2 < uncompressed.size());
+
+	StringRef verify = CompressionUtils::decompress(filter, compressed, arena);
+	ASSERT_EQ(verify.compare(uncompressed), 0);
 }
 
-#ifdef ZLIB_LIB_SUPPORTED
-TEST_CASE("/CompressionUtils/gzipCompression") {
+} // namespace
+
+TEST_CASE("/CompressionUtils/noCompression") {
 	Arena arena;
 	const int size = deterministicRandom()->randomInt(512, 1024);
 	Standalone<StringRef> uncompressed = makeString(size);
 	deterministicRandom()->randomBytes(mutateString(uncompressed), size);
 
-	Standalone<StringRef> compressed = CompressionUtils::compress(CompressionFilter::GZIP, uncompressed, arena);
-	ASSERT_NE(compressed.compare(uncompressed), 0);
+	Standalone<StringRef> compressed = CompressionUtils::compress(CompressionFilter::NONE, uncompressed, arena);
+	ASSERT_EQ(compressed.compare(uncompressed), 0);
 
-	StringRef verify = CompressionUtils::decompress(CompressionFilter::GZIP, compressed, arena);
+	StringRef verify = CompressionUtils::decompress(CompressionFilter::NONE, compressed, arena);
 	ASSERT_EQ(verify.compare(uncompressed), 0);
 
-	TraceEvent("GzipCompression_Done").log();
+	TraceEvent("NoCompressionDone");
+
+	return Void();
+}
+
+#ifdef ZLIB_LIB_SUPPORTED
+TEST_CASE("/CompressionUtils/gzipCompression") {
+	testCompression(CompressionFilter::GZIP);
+	TraceEvent("GzipCompressionDone");
 
 	return Void();
 }
 
 TEST_CASE("/CompressionUtils/gzipCompression2") {
-	Arena arena;
-	const int size = deterministicRandom()->randomInt(512, 1024);
-	std::string s(size, 'x');
-	Standalone<StringRef> uncompressed = Standalone<StringRef>(StringRef(s));
-	printf("Size before: %d\n", (int)uncompressed.size());
+	testCompression2(CompressionFilter::GZIP);
+	TraceEvent("GzipCompression2Done");
 
-	Standalone<StringRef> compressed = CompressionUtils::compress(CompressionFilter::GZIP, uncompressed, arena);
-	ASSERT_NE(compressed.compare(uncompressed), 0);
-	printf("Size after: %d\n", (int)compressed.size());
-	// Assert compressed size is less than half.
-	ASSERT(compressed.size() * 2 < uncompressed.size());
+	return Void();
+}
+#endif
 
-	StringRef verify = CompressionUtils::decompress(CompressionFilter::GZIP, compressed, arena);
-	ASSERT_EQ(verify.compare(uncompressed), 0);
+TEST_CASE("/CompressionUtils/zstdCompression") {
+	testCompression(CompressionFilter::ZSTD);
+	TraceEvent("ZstdCompressionDone");
+
+	return Void();
+}
 
-	TraceEvent("GzipCompression_Done").log();
+TEST_CASE("/CompressionUtils/gzipCompression2") {
```

Comment:
zstdCompression2?
I'm surprised something didn't happen weird with doubly registered test cases

Reply:
Thanks for pointing out, copy/paste typo.